modules/tinymce/src/plugins/accordion/main/ts/core/Utils.ts

```diff
@@ -0,0 +1,54 @@
+import Editor from 'tinymce/core/api/Editor';
+import Tools from 'tinymce/core/api/util/Tools';
+
+export const isSummary = (node?: Node | null): node is HTMLElement =>
+  node?.nodeName === 'SUMMARY';
+
+export const isDetails = (node?: Node | null): node is HTMLDetailsElement =>
+  node?.nodeName === 'DETAILS';
+
+export const isOpen = (details: HTMLDetailsElement): boolean =>
+  details.hasAttribute('open');
+
+export const isInSummary = (editor: Editor): boolean => {
+  const node = editor.selection.getNode();
+  return isSummary(node) || Boolean(editor.dom.getParent(node, isSummary));
+};
+
+export const getSelectedDetails = (editor: Editor): HTMLDetailsElement | null => {
```

Comment:
consider to return `Optional<HTMLDetailsElement>`

Reply:
It is much better with Optional. Thanks for the suggestion!